retrospective-update/functions.py

```
@@ -0,0 +1,351 @@
+
+    if not os.path.exists(params_file):
+        raise FileNotFoundError(f"Routing parameters file not found: {params_file}")
+    if not os.path.exists(connectivity_file):
+        raise FileNotFoundError(f"Connectivity file not found: {connectivity_file}")
+    if not os.path.exists(weight_table):
+        raise FileNotFoundError(f"Weight table file not found: {weight_table}")
+
+    output_dir = os.path.join(outputs_dir, vpu)
+    os.makedirs(output_dir, exist_ok=True)
+
+    for catchment_volumes_file in natsorted(glob(os.path.join(inflows_dir, os.path.basename(vpu_dir), '*.nc'))):
```

Ruff (0.12.2)

Undefined name `inflows_dir`

(F821)